pkg/cmd/secrets.go

```diff
@@ -380,6 +408,89 @@ func downloadSecrets(cmd *cobra.Command, args []string) {
 	utils.Log(fmt.Sprintf("Downloaded secrets to %s", filePath))
 }
 
+func templateSecrets(cmd *cobra.Command, args []string) {
+	localConfig := configuration.LocalConfig(cmd)
+
+	utils.RequireValue("token", localConfig.Token.Value)
+
+	filePath, err := utils.GetFilePath(args[0])
+	if err != nil {
+		utils.HandleError(err, "Unable to parse template file path")
+	}
+
+	var file []byte
+	file, err = ioutil.ReadFile(filePath)
+	if err != nil {
+		utils.HandleError(err, "Unable to read template file")
+	}
+
+	var outputFile *os.File
+	var outputFilePath string
+	output := cmd.Flag("output").Value.String()
+	if len(output) != 0 {
+		outputFilePath, err = utils.GetFilePath(output)
+		if err != nil {
+			utils.HandleError(err, "Unable to parse output file path")
+		}
+		outputFile, err = os.Create(outputFilePath)
+		if err != nil {
+			utils.HandleError(err, "Unable to open output file for writing")
+		}
+		defer outputFile.Close()
+	}
+
+	funcs := map[string]interface{}{
+		"tojson": func(value string) (string, error) {
+			body, err := json.Marshal(value)
+			if err != nil {
+				return "", err
+			}
+			return string(body), nil
+		},
+		"fromjson": func(value string) (interface{}, error) {
+			var result interface{}
+			err = json.Unmarshal([]byte(value), &result)
+			if err != nil {
+				return "", err
+			}
+			return result, nil
+		},
+	}
+	template, err := template.New("Secrets").Funcs(funcs).Parse(string(file))
+	if err != nil {
+		utils.HandleError(err, "Unable to parse template text")
+	}
+
+	response, responseErr := http.GetSecrets(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value)
+	if !responseErr.IsNil() {
+		utils.HandleError(responseErr.Unwrap(), responseErr.Message)
+	}
+
+	secrets, parseErr := models.ParseSecrets(response)
+	if parseErr != nil {
+		utils.HandleError(parseErr, "Unable to parse API response")
+	}
+
+	secretsMap := map[string]interface{}{}
+	for _, secret := range secrets {
+		secretsMap[secret.Name] = secret.ComputedValue
+	}
+
+	if outputFile != nil {
+		err = template.Execute(outputFile, secretsMap)
+		if err != nil {
+			utils.HandleError(parseErr, "Unable to render template")
```

Comment:
When testing locally using the provided template, my config didn't have the correct secrets. This was the output, which is a little confusing:

```
$ ./doppler secrets template template.yaml
host: <no value>
port: <no value>



Multiline: Unable to render template
Doppler Error: <nil>
```

Reply:
Yeah, I noticed this too. There is an option that I can add with `.Option("missingkey=error")` but if we do that, we won't be able to do `{{ if .MISSING_KEY }}` which I think could actually be really valuable.

Reply:
We could add a flag to enable/disable that. What do you think?

Reply:
Oh wait, I missed the partial write issue on my first read. Let me see what can be done about that